scene/src/realms/dungeon.ts

```diff
@@ -581,145 +505,40 @@ export class Dungeon implements Realm {
-      } else if (difficulty.HARD) {
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const hardEnemyAttack = 900
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const hardEnemyHP = 9000
-        // createCaveDungeonEnemies(hardEnemyAttack, hardEnemyHP)
-        // const wastelandApex = Array.from(
-        //     { length: 1 },
-        //     () => new HardCaveDungeonBoss(hardEnemyAttack)
-        // )
-        // wastelandApex.forEach((wastelandApex) => entityController.addEntity(wastelandApex))
-        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-      } else if (difficulty.NIGHTMARE) {
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const nightmareEnemyAttack = 2000
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const nightmareEnemyHP = 222500
-        //   createCaveDungeonEnemies(nightmareEnemyAttack, nightmareEnemyHP)
-        //   const wastelandApex = Array.from(
-        //     { length: 1 },
-        //     () => new NightmareCaveDungeonBoss(nightmareEnemyAttack)
-        // )
-        // wastelandApex.forEach((wastelandApex) => entityController.addEntity(wastelandApex))
-      } else {
-        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions, @typescript-eslint/no-base-to-string
-        console.log(`Invalid difficulty level: ${difficulty}`)
-      }
-    } else {
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      console.log(`Invalid location: ${location}`)
-    }
-  }
-
-  createDesertDungeonEnemies(attack: number, hp: number): void {
+  createDesertDungeonEnemies(gameController: GameController): void {
     desertSoldierPositions.forEach((position) => {
-      // this.createSoldier(position, attack, hp)
+      this.createSoldier(position, gameController)
     })
   }
 
-  createCaveDungeonEnemies(attack: number, hp: number): void {
-    caveSoldierPositions.forEach((position) => {
-      // this.createSoldier(position, attack, hp)
-    })
+  createDesertDungeonBoss(difficulty: string): void {
+    switch (difficulty) {
+      case 'easy':
+        // eslint-disable-next-line no-case-declarations, @typescript-eslint/no-unused-vars
+        const bossEasy = new EasyDesertDungeonBoss(8)
+        break
+      case 'medium':
+        // eslint-disable-next-line no-case-declarations, @typescript-eslint/no-unused-vars
+        const bossMed = new MedDesertDungeonBoss(80)
+        break
+      case 'hard':
+        // eslint-disable-next-line no-case-declarations, @typescript-eslint/no-unused-vars
+        const bossHard = new HardDesertDungeonBoss(300)
+        break
+      case 'nightmare':
+        // eslint-disable-next-line no-case-declarations, @typescript-eslint/no-unused-vars
+        const bossNightmare = new NightmareDesertDungeonBoss(800)
+        console.log('')
+    }
```

Comment:
We don't have the control of life-cycle of these bosses. Should we accumulate them in a list and clear it when changing realm/dungeon?